README.md

```diff
@@ -82,7 +91,17 @@ Write the BNF (not EBNF) description for this language.
 
 ## A2.1
 
-«replace this with your answer»
+<sentence> ::= "the" <description> <noun> <verb> <adverb> | "the" <description> <noun> <verb> | "the" <noun> <verb> <adverb> | "the" <noun> <verb>
+
+<description> ::= <adjective> <description> | <adjective>
+
+<adjective> ::= "lazy" | "smelly"
+
+<noun> ::= "dog" | "cat"
+
+<verb> ::= "ate" | "ran"
+
+<adverb> ::= "slowly" | "noisily"
```

Comment:
# 7

In general, you'd probably want to factor the optional things out of the <sentence> rule:

    <sentence> ::= "the" <adjectives> <noun> <verb> <optional-adverb>
    <adjectives> ::= <adjectives> <adjective> | «empty»
    <optional_adverb> ::= "slowly" | "noisily" | «empty»